larray/core.py

```diff
@@ -936,6 +936,56 @@ def labels(self, labels):
         self._labels = labels
         self._iswildcard = iswildcard
 
+    def extend(self, labels):
+        """
+        Append new labels to an axis or increase its length
+        in case of wildcard axis.
+        Note that `extend` does not occur in-place: a new axis
+        object is allocated, filled and returned.
+
+        Parameters
+        ----------
+        labels : int, iterable or Axis
+            * Number of new labels (only if wildcard axis).
+            * New labels to append to the axis.
+            * Axis to append. Must (not) be wildcard if self
+              is (not) wildcard.
```

Comment:
I guess you wanted to respect the order of the types, but I would rather have the second line "new labels to ..." first, because it is the most useful/usual option for users. In that case, to avoid confusion (with the types order) you could possibly not use an enumeration but a standard text. Something like:
```
labels : int, iterable or Axis
    New labels to append to the axis, possibly in the form of another Axis. If the current axis is a wildcard axis, passing a length is enough.